Solution

Shevtsooov opened the pull request:
[DEMO LINK](https://Shevtsooov.github.io/react_autocomplete/)

dinaminahi commented on src/App.tsx:
```diff
@@ -1,57 +1,63 @@
-          </div>
-        </div>
-      </div>
+      <Dropdown
+        query={query}
+        setQuery={setQuery}
+        applyQuery={applyQuery}
+        appliedQuery={appliedQuery}
+        filteredPeople={filteredPeople}
+        onSelected={selectPerson}
+        handleReset={handleReset}
+      />
```
Do we need all these props& or we can create some handlers?

dinaminahi reviewed (changes requested):
There is a bug we need to fix 

https://github.com/mate-academy/react_autocomplete/assets/46566190/9ceb190a-c8af-4635-bf9b-e0c4d74919bc

Shevtsooov commented:
@dinaminahi Hello, thanks for the detailed review.
I have fixed the bug and added handlers in my `Drodown` component, still I am not sure how to reduce the number of props.
Please suggest a solution.

dffUqp commented on src/components/Dropdown.tsx:
```diff
@@ -0,0 +1,83 @@
+            {filteredPeople.length > 0 && (
+              filteredPeople.map(person => (
+                <div className="dropdown-item">
+                  {// eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions
+                    <p
```
let's fix this eslint issue:smiley: 

I think we can replace `<p>` with `<a>`

dffUqp commented on src/components/Dropdown.tsx:
```diff
@@ -0,0 +1,83 @@
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          onChange={(event) => handleSearch(event)}
```
```suggestion
          onChange={handleSearch}
```

dffUqp commented on src/components/Dropdown.tsx:
```diff
@@ -0,0 +1,83 @@
+      {query && query === appliedQuery && (
+        <div className="dropdown-menu" role="menu">
+          <div className="dropdown-content">
+            {filteredPeople.length > 0 && (
```
do we really need this check?

dffUqp reviewed (approved):
GJ✔️

about props i think in you `App.tsx` you need only `selectedPerson` so the rest of the variables you can move to `DropDown.tsx`
